src/WidgetBase.ts

```diff
@@ -421,8 +464,8 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		return this._cachedVNode;
 	}
 
-	public invalidate(): void {
-		if (this._renderState === WidgetRenderState.IDLE) {
+	public invalidate(force?: boolean): void {
```

Comment:
Can you explain why this is being added as part of this PR?  It doesn't feel germane to the intent of the PR, but maybe I am missing something.